### ISSUE TITLE
Conversion functions

### ISSUE THREAD

**Issue by archseer** [archseer/enigma#32]
It's time to get all these out of the way.

#### x_to_list
- [X] erlang:atom_to_list/1
- [X] erlang:binary_to_list/1
- [ ] erlang:binary_to_list/3
- [ ] erlang:float_to_list/1
- [ ] erlang:float_to_list/2
- [X] erlang:integer_to_list/1
- [ ] erlang:integer_to_list/2
- [X] erlang:pid_to_list/1
- [X] erlang:tuple_to_list/1
- [X] erlang:ref_to_list/1
- [ ] erlang:port_to_list/1
- [X] erlang:fun_to_list/1
- [ ] erlang:bitstring_to_list/1

#### list_to_x

- [X] erlang:list_to_atom/1
- [X] erlang:list_to_binary/1
- [ ] erlang:list_to_float/1
- [X] erlang:list_to_integer/1
- [ ] erlang:list_to_pid/1
- [ ] erlang:list_to_port/1
- [ ] erlang:list_to_ref/1
- [X] erlang:list_to_tuple/1
- [ ] string:list_to_integer/1
- [ ] erlang:list_to_integer/2
- [ ] string:list_to_float/1
- [ ] erlang:list_to_existing_atom/1
- [ ] erlang:list_to_bitstring/1

#### binary_to_x
- [ ] erlang:binary_to_float/1
- [X] erlang:binary_to_term/1
- [ ] erlang:binary_to_atom/2
- [ ] erlang:binary_to_existing_atom/2
- [ ] erlang:binary_to_integer/1
- [ ] erlang:binary_to_integer/2

#### x_to_binary

- [ ] erlang:atom_to_binary/2
- [ ] erlang:float_to_binary/1
- [ ] erlang:float_to_binary/2
- [ ] erlang:integer_to_binary/1
- [ ] erlang:integer_to_binary/2
- [ ] erlang:term_to_binary/1
- [ ] erlang:term_to_binary/2

#### Special

- [ ] unicode:characters_to_binary/2
- [X] unicode:characters_to_list/2
- [X] erlang:iolist_to_binary/1
- [ ] erlang:iolist_to_iovec/1

**Comment by cgmcintyr** [comment]
Hi, is anyone working on these at the moment?

**Comment by archseer** [comment]
Nope, feel free to go ahead :) I have some changes pending to update us to latest rust nightly, but I ran into some regressions I need to report upstream first.
